wikipedia/scripts/fetch_enwiki_revisions.py

```diff
@@ -0,0 +1,156 @@
+    parser.add_argument('-d', '--query_date', help='Date if not yesterday, in YYYYMMDD format.', type=str)
+    parser.add_argument('-L', '--logging_level', help='Logging level. Options are debug, info, warning, error, critical. Default: info.', default='info', type=str), 
+    parser.add_argument('-W', '--logging_destination', help='Logging destination file. (default: standard error)', type=str), 
+    args = parser.parse_args()
+    return(args)
+
+def main():
+    args = parse_args()
+
+    output_path = args.output_folder
+    article_filename = args.article_file
+    #handle -d
+    if args.query_date:
+        query_date = args.query_date
+    else:
+        yesterday = datetime.datetime.today() - datetime.timedelta(days=1)
+        query_date = yesterday.strftime("%Y%m%d")
+
+    query_data  = query_date + "00" #requires specifying hours
+
+    #handle -L
+    loglevel_mapping = { 'debug' : logging.DEBUG,
+                         'info' : logging.INFO,
+                         'warning' : logging.WARNING,
+                         'error' : logging.ERROR,
+                         'critical' : logging.CRITICAL }
+
+    if args.logging_level in loglevel_mapping:
+        loglevel = loglevel_mapping[args.logging_level]
+    else:
+        print("Choose a valid log level: debug, info, warning, error, or critical") 
+        exit
+
+    #handle -W
+    if args.logging_destination:
+        logging.basicConfig(filename=args.logging_destination, filemode='a', level=loglevel)
+    else:
+        logging.basicConfig(level=loglevel)
+
+    export_git_hash = subprocess.check_output(['git', 'rev-parse', 'HEAD']).decode().strip()
+    export_git_short_hash = subprocess.check_output(['git', 'rev-parse', '--short', 'HEAD']).decode().strip()
+    export_time = str(datetime.datetime.now())
+
+    logging.info(f"Starting run at {export_time}")
+    logging.info(f"Last commit: {export_git_hash}")
+
+    json_output_filename = os.path.join(output_path, f"digobs_covid19-wikipedia-enwiki_revisions-{query_date}.json")
+    tsv_output_filename =  os.path.join(output_path, f"digobs_covid19-wikipedia-enwiki_revisions-{query_data}.tsv")
+    
+    api_session = api.Session("https://en.wikipedia.org/w/api.php")
+
+    # list of properties from the API we want to gather (basically all of
+    # them supported by mediawik-utilities)
+
+    rv_props =  {'revid' : 'ids',
+                 'timestamp' : 'timestamp',
+                 'user' : 'user',
+                 'userid' : 'userid',
+                 'size' : 'size',
+                 'sha1' : 'sha1',
+                 'contentmodel' : 'contentmodel',
+                 'tags' : 'tags',
+                 'comment' : 'comment',
+                 'content' : 'content' }
+
+    exclude_from_tsv = ['tags', 'comment', 'content']
+
+    # load the list of articles
+    with open(article_filename, 'r') as infile:
+        article_list = [art.strip() for art in list(infile)]
```

Comment:
This does two passes over infile. First in the call to list, then in the listcomp

i would do:
list(map(str.strip, infile))